### PR TITLE
Feed connector

### PR DESCRIPTION
**Pull request by hmrajas**

### CODE REVIEW

File: feedconnector/feedconnector/org.wso2.carbon.connector/pom.xml [hunk 1]
```diff
@@ -0,0 +1,884 @@
...
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.securevault</artifactId>
+            <version>4.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.equinox</groupId>
+            <artifactId>org.eclipse.equinox.http.helper</artifactId>
+            <version>1.1.0.wso2v1</version>
+        </dependency>
+    </dependencies>
```
Comment (RKathees) [inline feedconnector/feedconnector/org.wso2.carbon.connector/pom.xml]: The required dependencies are already added in the parent pom. Please add specific dependencies only in this pom and move the common dependencies to parent pom. Each dependency version should be moved to the properties. 
https://github.com/wso2/esb-connectors/blob/master/pom.xml

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedConstant.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public static final String title = "Title";
+    public static final String content = "Content";
+    public static final String author = "Author";
+    public static final String feedID = "FeedID";
+    public static final String contentType = "application/atom+xml;type=entry";
+    public static final String Result = "Result";
+    public static final String Status = "status";
+    public static final String result = "result";
+    public static final String entryID = "EntryID";
+    public static final String ns = "ns";
```
Comment (RKathees) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedConstant.java]: The constant variable names should be all caps.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/pom.xml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    <url>http://wso2.org</url>
+    <distributionManagement>
+        <repository>
+            <id>wso2-maven2-repository</id>
+            <name>WSO2 Maven2 Repository</name>
+            <url>scp://dist.wso2.org/home/httpd/dist.wso2.org/maven2/</url>
+        </repository>
+    </distributionManagement>
+    <repositories>
+        <repository>
```
Comment (Shakila) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/pom.xml]: This repository is already added in parent pom. So we can remove from here.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedUpdation.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            handleException("Entry ID and host address can not be null or empty", messageContext);
+        }
+        String entryUri = hostAddress + "/" + entryID + "-";
+        AbderaClient abderaClient = FeedUtil.getAbderaClient();
+        try {
+            doc = abderaClient.get(entryUri).getDocument();
+        } catch (Exception e) {
+            handleException("error while get the entry from server ", e, messageContext);
+        }
+        if (title != null) {
```
Comment (Shakila) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedUpdation.java]: Use StringUtils for null check.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    public void connect(MessageContext messageContext) throws ConnectException {
+        String hostAddress = (String) getParameter(messageContext, FeedConstant.HOST_ADDRESS);
+        String title = (String) getParameter(messageContext, FeedConstant.TITLE);
+        String content = (String) getParameter(messageContext, FeedConstant.CONTENT);
+        String author = (String) getParameter(messageContext, FeedConstant.AUTHOR);
+        String feedID = (String) getParameter(messageContext, FeedConstant.FEED_ID);
+
+        if (StringUtils.isEmpty(hostAddress)) {
+            handleException("host address can not be null or empty", messageContext);
```
Comment (keerthu) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java]: Use capital letter at the start of the error message.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        RequestOptions opts = new RequestOptions();
+        opts.setContentType(FeedConstant.CONTENT_TYPE);
+        FeedUtil response = new FeedUtil();
+        ClientResponse resp;
+        try {
+            resp = abderaClient.post(hostAddress, entry, opts);
+            response.InjectMessage(messageContext, resp.getStatusText());
+        } catch (Exception ex) {
+            handleException("error while connect ", ex, messageContext);
```
Comment (keerthu) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java]: Use capital letter at the start. Check in all files.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/pom.xml (hunk 3 above)
Comment (hmrajas) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/pom.xml]: Fixed in https://github.com/wso2/esb-connectors/pull/551

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedUpdation.java (hunk 4 above)
Comment (hmrajas) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedUpdation.java]: Fixed in https://github.com/wso2/esb-connectors/pull/551

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java (hunk 5 above)
Comment (hmrajas) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java]: Fixed in https://github.com/wso2/esb-connectors/pull/551

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java (hunk 6 above)
Comment (hmrajas) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FeedCreation.java]: Fixed in https://github.com/wso2/esb-connectors/pull/551

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (esb-connectors) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml]: Add copy rights header in all file.

File: feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
```
Comment (RKathees) [inline feed/feed-connector/feed-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml]: Add copy rights header.

**Pull request merged**